packages/truffle-debugger/lib/data/reducers.js

```diff
@@ -187,30 +196,104 @@ function mappingKeys(state = DEFAULT_MAPPING_KEYS, action) {
       );
       return {
         decodingStarted: state.decodingStarted + (action.started ? 1 : -1),
-        byId: { ...state.byId }
+        byAddress: state.byAddress
       };
-    case actions.MAP_KEY:
-      let { id, key } = action;
-      debug("mapping id and key: %s, %o", id, key);
+    case actions.MAP_PATH_AND_ASSIGN:
```

Comment:
This case is difficult to reason about at a high-level. Might be worth a couple sentence comment explaining the general approach.

Reply:
OK, I added one!  Hopefully you find it sufficiently explanatory.  (Honestly it's possible that the code in this case can be simplified now that I took out the relinking system -- as distinct from the initial linking system that's still in place -- but I didn't want to think through it again. :P )

Reply:
looks good!